tests/_utils.py

```diff
@@ -54,7 +65,147 @@ def create_test_tracklet(track_len: int):
     return tracklet, data, properties, track_ID
 
 
-def full_tracker_example(objects):
+def create_realistc_tracklet(
+    start_x: float,
+    start_y: float,
+    dx: float,
+    dy: float,
+    track_len: float,
+    track_ID: int,
+) -> btrack.btypes.Tracklet:
+    """Create a realistic moving track."""
+    data = {
+        "x": start_x,
+        "y": start_y,
+        "z": 0,
+        "t": 0,
+        "ID": (track_ID - 1) * track_len,
+    }
+
+    objects = [
+        btrack.btypes.PyTrackObject.from_dict(data),
+    ]
+
+    for t in range(1, track_len):
+        new_data = {
+            "x": data["x"] + dx,
+            "y": data["y"] + dy,
+            "t": t,
+            "ID": (track_ID - 1) * track_len + t,
+        }
+        data.update(new_data)
+        objects.append(
+            btrack.btypes.PyTrackObject.from_dict(data),
+        )
+
+    return btrack.btypes.Tracklet(track_ID, objects)
+
+
+def create_test_image(
+    boxsize: int = 150,
+    ndim: int = 2,
+    nobj: int = 10,
+    binsize: int = 5,
+    binary: bool = True,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Make a test image that ensures that no two pixels are in contact."""
+    shape = (boxsize,) * ndim
+    img = np.zeros(shape, dtype=np.uint16)
+
+    # return an empty image if we have no objects
+    if nobj == 0:
+        return img, None
+
+    # split this into voxels
+    bins = boxsize // binsize
+
+    def _sample():
+        _img = np.zeros((binsize,) * ndim, dtype=np.uint16)
+        _coord = tuple(
+            np.random.randint(1, binsize - 1, size=(ndim,)).tolist()
+        )
+        _img[_coord] = 1
+        assert np.sum(_img) == 1
+        return _img, _coord
+
+    # now we update nobj grid positions with a sample
+    grid = np.stack(np.meshgrid(*[np.arange(bins)] * ndim), -1).reshape(
+        -1, ndim
+    )
+    rng = np.random.default_rng(seed=1234)
```

Comment:
I would suggest that somewhere we make a seed variable and import it everywhere we use it. Looks like we're now using it 3 times